lib/client.js

```diff
@@ -14,64 +14,102 @@ module.exports = function (dependencies) {
     HTTP2_HEADER_METHOD,
     HTTP2_HEADER_AUTHORITY,
     HTTP2_HEADER_PATH,
-    HTTP2_METHOD_POST
+    HTTP2_METHOD_POST,
+    NGHTTP2_CANCEL,
   } = http2.constants;
 
   function Client (options) {
     this.config = config(options);
     this.healthCheckInterval = setInterval(() => {
-         if (this.session && !this.session.destroyed) {
-             this.session.ping((error, duration) => {
-                 if (error) {
-                     logger("No Ping response after " + duration + " ms");
-                 }
-                 logger("Ping response after " + duration + " ms");
-             })
-         }
-     }, this.config.heartBeat).unref();
+      if (this.session && !this.session.closed && !this.session.destroyed && !this.isDestroyed) {
+        this.session.ping((error, duration) => {
+          if (error) {
+            logger("No Ping response after " + duration + " ms with error:" + error.message);
+            return;
+          }
+          logger("Ping response after " + duration + " ms");
+        });
+      }
+    }, this.config.heartBeat).unref();
+  }
+
+  Client.prototype.destroySession = function (session, callback) {
```

Comment:
Could document here and in closeAndDestroySession that session should be set except when destroying the client